### ISSUE TITLE
Segfault in current master

### ISSUE THREAD

**Issue by prolic** [pdezwart/php-amqp#221]
I can create a segmentation fault with this simple script:

```<?php

$connection = new \AMQPConnection();
$connection->connect();

$channel = new \AMQPChannel($connection);

$exchange = new \AMQPExchange($channel);

try {
    $exchange->delete();
} catch (\Exception $e) {
    // ignore
}```

**Comment by prolic** [comment]
My system:
PHP 7.0.3
Linux 4.2.0-30-generic Ubuntu
with php-amqp extension from current master

**Comment by pinepain** [comment]
Hi, I'll give it a look later today.

**Comment by pinepain** [comment]
Can't reproduce from first try in php-amqp vagrant box.  It's not exactly your configuration, so see versions below. Can you try to reproduce segfault in clean environment or post some stack trace please?

```
$ php -v
PHP 7.0.3 (cli) (built: Feb 16 2016 11:38:45) ( NTS DEBUG )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
```

```
$ php --ri amqp

amqp

Version => v1.7.0-master-dev
Revision => 7960bdd
Compiled => May 16 2016 @ 17:59:48
AMQP protocol version => 0-9-1
librabbitmq version => 0.7.1
Default max channels per connection => 256
Default max frame size => 131072
Default heartbeats interval => 0

Directive => Local Value => Master Value
amqp.host => localhost => localhost
amqp.vhost => / => /
amqp.port => 5672 => 5672
amqp.timeout => no value => no value
amqp.read_timeout => 0 => 0
amqp.write_timeout => 0 => 0
amqp.connect_timeout => 0 => 0
amqp.login => guest => guest
amqp.password => guest => guest
amqp.auto_ack => 0 => 0
amqp.prefetch_count => 3 => 3
amqp.channel_max => 256 => 256
amqp.frame_max => 131072 => 131072
amqp.heartbeat => 0 => 0
amqp.cacert => no value => no value
amqp.cert => no value => no value
amqp.key => no value => no value
amqp.verify => 1 => 1
```

```
$ sudo rabbitmqctl status 
Status of node 'rabbit@vagrant-ubuntu-wily-64' ...
[{pid,1801},
 {running_applications,
     [{rabbitmq_management,"RabbitMQ Management Console","3.6.0"},
      {rabbitmq_web_dispatch,"RabbitMQ Web Dispatcher","3.6.0"},
      {webmachine,"webmachine","git"},
      {mochiweb,"MochiMedia Web Server","2.13.0"},
      {rabbitmq_management_agent,"RabbitMQ Management Agent","3.6.0"},
      {rabbit,"RabbitMQ","3.6.0"},
      {ssl,"Erlang/OTP SSL application","7.0"},
      {public_key,"Public key infrastructure","1.0"},
      {crypto,"CRYPTO","3.6"},
      {os_mon,"CPO  CXC 138 46","2.4"},
      {asn1,"The Erlang ASN1 compiler version 4.0","4.0"},
      {amqp_client,"RabbitMQ AMQP Client","3.6.0"},
      {xmerl,"XML parser","1.3.8"},
      {inets,"INETS  CXC 138 49","6.0"},
      {ranch,"Socket acceptor pool for TCP protocols.","1.2.1"},
      {compiler,"ERTS  CXC 138 10","6.0"},
      {mnesia,"MNESIA  CXC 138 12","4.13"},
      {rabbit_common,[],"3.6.0"},
      {syntax_tools,"Syntax tools","1.7"},
      {sasl,"SASL  CXC 138 11","2.5"},
      {stdlib,"ERTS  CXC 138 10","2.5"},
      {kernel,"ERTS  CXC 138 10","4.0"}]},
 {os,{unix,linux}},
 {erlang_version,
     "Erlang/OTP 18 [erts-7.0] [source] [64-bit] [async-threads:64] [kernel-poll:true]\n"},
 {memory,
     [{total,53318944},
      {connection_readers,0},
      {connection_writers,0},
      {connection_channels,0},
      {connection_other,2592},
      {queue_procs,183784},
      {queue_slave_procs,0},
      {plugins,318528},
      {other_proc,18700064},
      {mnesia,79712},
      {mgmt_db,153144},
      {msg_index,56336},
      {other_ets,1373872},
      {binary,31736},
      {code,27187615},
      {atom,992409},
      {other_system,4239152}]},
 {alarms,[]},
 {listeners,[{clustering,25672,"::"},{amqp,5672,"::"},{'amqp/ssl',5671,"::"}]},
 {vm_memory_high_watermark,0.4},
 {vm_memory_limit,839234355},
 {disk_free_limit,50000000},
 {disk_free,38294716416},
 {file_descriptors,
     [{total_limit,924},{total_used,2},{sockets_limit,829},{sockets_used,0}]},
 {processes,[{limit,1048576},{used,217}]},
 {run_queue,0},
 {uptime,938},
 {kernel,{net_ticktime,60}}]
```

```
$ uname -a
Linux vagrant-ubuntu-wily-64 4.2.0-23-generic #28-Ubuntu SMP Sun Dec 27 17:47:31 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 15.10
Release:	15.10
Codename:	wily
```

**Comment by prolic** [comment]
Here is my gdb backtrace

    Program terminated with signal SIGSEGV, Segmentation fault.
    #0  strlen () at ../sysdeps/x86_64/strlen.S:106
    106     ../sysdeps/x86_64/strlen.S: Datei oder Verzeichnis nicht gefunden.
    (gdb) bt
    #0  strlen () at ../sysdeps/x86_64/strlen.S:106
    #1  0x00007f9e19390459 in amqp_cstring_bytes (cstr=0x45 <error: Cannot access memory at address 0x45>)
        at librabbitmq/amqp_mem.c:183
    #2  0x00007f9e195a61df in zim_amqp_exchange_class_delete (execute_data=0x7f9e19c12180, return_value=0x7f9e19c12170)
        at /home/sasa/code/php-amqp/amqp_exchange.c:413
    #3  0x00000000008b6db2 in ZEND_DO_FCALL_SPEC_HANDLER () at /home/sasa/Downloads/php-7.0.3/Zend/zend_vm_execute.h:842
    #4  0x000000000086c74b in execute_ex (ex=<optimized out>) at /home/sasa/Downloads/php-7.0.3/Zend/zend_vm_execute.h:414
    #5  0x00000000008c7bb9 in zend_execute (op_array=<optimized out>, return_value=<optimized out>)
        at /home/sasa/Downloads/php-7.0.3/Zend/zend_vm_execute.h:458
    #6  0x0000000000828e17 in zend_execute_scripts (type=type@entry=8, retval=retval@entry=0x0, 
        file_count=file_count@entry=3) at /home/sasa/Downloads/php-7.0.3/Zend/zend.c:1427
    #7  0x00000000007ba758 in php_execute_script (primary_file=primary_file@entry=0x7ffde4a01cd0)
        at /home/sasa/Downloads/php-7.0.3/main/main.c:2484
    #8  0x00000000008c9ad4 in do_cli (argc=2, argv=0x1ae1c20) at /home/sasa/Downloads/php-7.0.3/sapi/cli/php_cli.c:974
    #9  0x000000000043c5d2 in main (argc=2, argv=0x1ae1c20) at /home/sasa/Downloads/php-7.0.3/sapi/cli/php_cli.c:1345

**Comment by pinepain** [comment]
I guess this is the same error like #226 and should be fixed by #230. Can you please verify?

**Comment by lstrojny** [comment]
Closing
